### PR TITLE
Use OAuth2 on GitHub

### PR DESCRIPTION
**Pull request by sbrunner**
Weakness:
- shared secret
- no brut force protection

@gberaudo Do you agree with that?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,8 +8,9 @@
 - Remove the `development.ini` file to simplify the default application template; restore `production.ini` has the default configuration file.
 - When the `auth_type` is GitHub the `is_auth` and the `has_access` can return different values, the `is_auth` will just check
   the authentication on GitHub succeed, `has_access` also check that the user has the desired rights.
+- `C2C_SECRET` became deprecated, the GitHub login should be use instead, because he is relay safer.
```
Comment (arnaud-morvan) [inline CHANGELOG.md:11]:
```suggestion
- `C2C_SECRET` became deprecated, recommend authentication method is Oauth2 with GitHub for security reason.
```

**Review by arnaud-morvan — APPROVED** [review]

**Comment by gberaudo** [comment]
@sbrunner, for me deprecating C2C_SECRET is going too far. C2cwsgiutils users should be able to continue to choose this authentication method in the future.

**Comment by sbrunner** [comment]
@gberaudo If we want to keep it, we should do something else like
adding a warning message in login part to discourage to use it in production...

For me, we can add a deprecated warning now
and then think what we will do with that in the future...

**Review by gberaudo — APPROVED** [review]
@sbrunner, do you want also to add a message when the app is starting up?

**Pull request merged**